src/core_plugins/kibana/public/management/sections/indices/create_index_pattern/create_index_pattern.js

```diff
@@ -207,39 +195,54 @@ uiModules.get('apps/management')
     return loadingCount > 0;
   };
 
-  this.refreshFieldList = () => {
-    const timeField = this.formValues.timeField;
+  this.refreshTimeFieldOptions = () => {
+    const prevOption = this.formValues.timeFieldOption;
+
     loadingCount += 1;
-    fetchFieldList().then(results => {
-      if (timeField) {
-        updateFieldListAndSetTimeField(results, timeField.name);
-      } else {
-        updateFieldList(results);
-      }
-    }).finally(() => {
-      loadingCount -= 1;
-    });
+    this.timeFieldOptions = [];
+    this.timeFieldOptionsError = null;
+    this.formValues.timeFieldOption = null;
+    getTimeFieldOptions()
```

Comment:
>  I'm very interested in understanding the concerns around explicit `$scope.$apply()`. I've always considered it a valid option and the best option when dealing with async operations outside of angular.

I agree that sometimes `$scope.$apply()` is the only way to inter-op between angular and non-angular async contexts. My concern is that using `$scope.$apply()` when not absolutely necessary leads to trouble. We did this in Kibana 3 and ended up with more than a couple PR's that just converted a `$scope.$apply()` to `if (!$scope.$$phase) $scope.$apply()` because things started to get out of hand. I'm just saying that if we can avoid using `$scope.$apply()` (like by using native angular promises) then I think we should.